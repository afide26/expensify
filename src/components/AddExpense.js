import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expensesActions';


export class AddExpensePage extends Component{
  constructor(props){
    super(props);
  };

  render(){
    return(
    <div>
      <h1>Add Expense</h1>
      <ExpenseForm
        buttonLabel="Add"
        onSubmit = {(expense)=>{
          this.props.addExpense(expense);
          this.props.history.push('/')
        }}
      />
    </div>   
    )
  }
}

const mapDispatchToProps = (dispatch)=>{
  return{
    addExpense: (expense)=> dispatch(addExpense(expense))
  }
}
export default connect(undefined, mapDispatchToProps)(AddExpensePage);