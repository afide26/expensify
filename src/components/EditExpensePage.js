import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense,removeExpense } from '../actions/expensesActions';

export class EditExpensePage extends Component{
  onSubmit = (expense)=>{
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = ()=>{
    this.props.removeExpense({id:this.props.expense.id})
    this.props.history.push('/');
  };

  render(){
    return(
      <div className="edit-expense">
        <ExpenseForm 
            buttonLabel = "Edit" 
            expense = {this.props.expense}
            onSubmit={this.onSubmit}
        />
        <div className="edit-expense__remove">
          <button
            className="edit-expense--button"
            onClick={this.onRemove}
          >
            Remove
          </button>
        </div>
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch, props)=>{
  return{
    editExpense: (id,expense) => dispatch(editExpense(id,expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
  }
}
const mapStateToProps = (state,props)=>{
  return(
    {
      expense:state.expenses.find((expense)=> expense.id === props.match.params.id)
    }
  )
}
export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);