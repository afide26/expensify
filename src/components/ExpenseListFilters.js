import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filtersActions';

export class ExpenseListFilters extends Component{
  state={
    calendarFocused: null
  };

  onDatesChange = ({startDate, endDate})=>{
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }

  onFocusChange = (calendarFocused)=>{
    this.setState(()=>({calendarFocused}))
  }

  onTextChange = (e)=> {
    const text = e.target.value;
    this.props.setTextFilter(text);
  }

  onSortChange = (e)=>{
    if(e.target.value === 'date'){
      this.props.sortByDate();
    }else if( e.target.value === 'amount'){
      this.props.sortByAmount();
    }
  }

  render(){
    return(
      <div>
        <label htmlFor="search">Search:</label> 
        <input 
          name="search"
          type="text" 
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <label htmlFor="sort">Sort By:</label>
        <select 
          name="sort" 
          value={this.props.filters.sortBy}
          onChange = {this.onSortChange}
          >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput ={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          showClearDates={true}
          isOutsideRange={()=> false}
        />
      </div>
    )
  }
}


const mapStateToProps = (state)=>({filters:state.filters})

const mapDispatchToProps = (dispatch)=>(
  {
    setTextFilter:(text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate)=> dispatch(setEndDate(endDate)),
  }
)


export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);

