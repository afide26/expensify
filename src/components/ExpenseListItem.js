import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expensesActions';

const ExpenseListItem = ({dispatch, id, description,note, amount, createdAt})=>{
  return(
    <div>
      <h3><strong>{description}</strong></h3>
      <p>Amount due: ${amount/100}.00</p>
      <p>Note: <em>{note}</em></p>
      <code>Created at: {createdAt}</code>
      <button onClick={()=>(dispatch(removeExpense({id})))}>
        Remove
      </button>
      <hr/>
    </div>
  )
}


export default connect()(ExpenseListItem);