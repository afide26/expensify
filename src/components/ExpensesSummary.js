import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import totalExpenses from './../selectors/expenses-total';
import selectedExpenses from './../selectors/expenses';


export class ExpensesSummary extends Component {
  expenseTerm = (expenses)=>{
    return expenses.length === 0 || expenses.length === 1
           ? 'expense'
           : 'expenses'
  }
  render(){
    return(
      <div>
        <p>You are viewing - {this.props.expenses.length} {this.expenseTerm(this.props.expenses)} </p>
        <p>The total amount of all expenses is - {numeral(this.props.totalExpenses).format('$0,0.00')}</p>
      </div>
    )
  }
}
const mapStateToProps = (state)=>(
  {
    totalExpenses: totalExpenses(selectedExpenses(state.expenses, state.filters)),
    expenses: selectedExpenses(state.expenses, state.filters)
  }
)

export default connect(mapStateToProps)(ExpensesSummary);