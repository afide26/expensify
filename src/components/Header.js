import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = ()=>(
  <header className="header">
    <img src="../../images/AFavicon.png" className="header__image"/>
    <h1 className="header__title">Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
  </header>
)


export default Header;