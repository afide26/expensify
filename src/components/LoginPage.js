import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from './../actions/Auth';

export const LoginComponent = ({startLogin})=>{
  return(
    <div className="login">
      <form name="Login" className="login">
        <button onClick={startLogin}>Login with Google</button>
      </form> 
    </div>
  )
}


const mapDispatchToProps = (dispatch)=>{
  return{
    startLogin: ()=> dispatch(startLogin())
  }
};

export default connect(undefined,mapDispatchToProps)(LoginComponent)