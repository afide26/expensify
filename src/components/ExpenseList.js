import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectedExpenses from '../selectors/expenses';

const ExpenseList = (props)=>(
  <div>
    <h2>Expense List - {props.expenses.length>0 ? props.expenses.length : <small><em>No records found</em></small>} </h2>
    <ul>
      <li>
        {props.expenses.map((expense)=>(
          <ExpenseListItem 
            key={expense.id} 
            {...expense}
          />
        ))}
      </li>
    </ul>
  </div>
);

const mapStateToProps = (state)=>(
  {
    expenses:selectedExpenses(state.expenses, state.filters)
  }
)

export default connect(mapStateToProps)(ExpenseList)

