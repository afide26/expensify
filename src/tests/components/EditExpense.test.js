import React from 'react';
import { EditExpensePage } from '../../components/EditExpensePage';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expense.fixture';

let startEditExpense, history, startRemoveExpense, wrapper;

beforeEach(()=>{
  startEditExpense= jest.fn();
  history= {push:jest.fn()};
  startRemoveExpense = jest.fn();
  wrapper = shallow(<EditExpensePage 
                      history={history} 
                      startEditExpense={startEditExpense} 
                      startRemoveExpense={startRemoveExpense}
                      expense={expenses[0]}
                      />);
})

test('should render EditExpensePage', ()=>{
  expect(wrapper).toMatchSnapshot();
});

test("should handle EditExpense", ()=>{
  const expense = expenses[0]
  wrapper.find('ExpenseForm').prop('onSubmit')(expense);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startEditExpense).toHaveBeenLastCalledWith(expense.id, expense);
});

test("should handle RemoveExpense",()=>{
  const expense = expenses[0];
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startRemoveExpense).toHaveBeenLastCalledWith({id:expense.id})
});
