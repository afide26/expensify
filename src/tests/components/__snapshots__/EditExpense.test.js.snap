// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div
  className="edit-expense"
>
  <ExpenseForm
    buttonLabel="Edit"
    expense={
      Object {
        "amount": 395,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "now",
      }
    }
    onSubmit={[Function]}
  />
  <div
    className="edit-expense__remove"
  >
    <button
      className="edit-expense--button"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
